File: proj04/Factorial.java
/* Factorial.java calculates the factorial of a given natural number.
 *
 * Input: n, a natural number.
 * PRE: 0 <= n
 * Output: n factorial.
 * By: Shion Fukuzawa, for CS 214 at Calvin College. 
 * Date: February 23, 2018
 ********************************************************/
import java.util.Scanner; 		// Get input

class Factorial {

  /* function factorial() returns the factorial of n.
      * Parameter: n, a natural number
      * Precondition: 0 <= n.
      * Returns: n factorial.
      */
    public static int factorial(int n) {
      int res = 1;  
      for (int i = 1; i <= n; i++) {
          res *= i;
      }               
      return res; // return result
    } // factorial method

  public static void main(String[] args){
    Scanner keyboard = new Scanner(System.in);
    System.out.println("To compute n!, enter n:");
    int n;
    n = keyboard.nextInt();
    System.out.printf("%d! = %d\n", n, factorial(n));
  }
}

File: proj04/factorial.rb
#! /usr/bin/ruby
# factorial.rb computes the factorial of a given integer n
# 
# Input: a non-negative integer n
# Precondition: n not negative
# Output: n!
#
# By: Shion Fukuzawa, for CS214 at Calvin College
# Date: February 23, 2018
###############################################################

# function factorial returns a n factorial, given integer n
# Parameters: n, a number
# Precondition: n >= 0.
# Returns: n!
def factorial(n)
    ret = 1
    for i in 1..n 
        ret *= i
    end
    return ret
end

if __FILE__ == $0
   puts "To computer n!, "
   print " enter n: "
   n = gets.chomp.to_i
   print n
   print "! = "
   puts factorial(n)
end
